apps/minifront/src/components/swap/auction-list/claim-all-button.tsx

```diff
@@ -0,0 +1,117 @@
+import { useAuctionInfos } from '../../../state/swap/dutch-auction';
+import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import { Button } from '@repo/ui/components/ui/button';
+import { AllSlices } from '../../../state';
+import { useStoreShallow } from '../../../utils/use-store-shallow.ts';
+import { filterWithLimit } from './helpers.ts';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@repo/ui/components/ui/tooltip';
+import { AuctionInfo } from '../../../fetchers/auction-infos.ts';
+
+const claimAllAuctionButtonSelector = (state: AllSlices) => ({
+  endAllAuctions: state.swap.dutchAuction.endAllAuctions,
+  withdrawAllAuctions: state.swap.dutchAuction.withdrawAllAuctions,
+});
+
+export interface AuctionsBatch {
+  auctions: AuctionInfo[];
+  source: AddressIndex;
+}
+
+// Assemble batch auctions for end or withdrawal.
+// All auctions in the batch will have the same 'AddressIndex'
+export const assembleAuctionBatch = (
+  auctions: AuctionInfo[],
+  filteredSeqNumber: bigint,
+  batchLimit: number,
+): AuctionsBatch => {
```

Comment:
This is a temporary solution that allows you to close all auctions on accounts with different batches
If a user has auctions on **account 0** and **account 1,** the first transaction will end all auctions on **account 0**, and the second transaction will end all auctions on **account 1**